src/org/zaproxy/zap/extension/scripts/CommandPanel.java

```diff
@@ -97,21 +114,28 @@ public void addKeyListener(KeyListener l) {
 	}
 	
 	public void setSyntax (String syntax) {
-		getTxtOutput().setSyntaxEditingStyle(syntax);
+		boolean isHighlightEnabled = getTxtOutput().getDocument().getLength() < HIGHLIGHT_SCRIPT_MAX_SIZE_THRESHOLD && !isLargeScriptContent;
+		getTxtOutput().setSyntaxEditingStyle(isHighlightEnabled ? syntax : SyntaxConstants.SYNTAX_STYLE_NONE);
 	}
 
 
 	public void clear() {
-	    getTxtOutput().setText("");
+		setCommandScriptContent(null);
 	    getTxtOutput().discardAllEdits();
 	    setSyntax(SyntaxConstants.SYNTAX_STYLE_NONE);
 	}
 
 	public String getCommandScript() {
-		return getTxtOutput().getText();
+		return isLargeScriptContent ? largeScriptContent : getTxtOutput().getText();
 	}
-	
-	protected void appendToCommandScript (String str) {
+
+	public void setCommandScript(String str) {
+		setCommandScriptContent(str);
+		getTxtOutput().discardAllEdits();
+		getTxtOutput().requestFocus();
+	}
+
+	protected void appendToCommandScript(String str) {

```

Comment:
Since this is being called "always" before a `clear` it makes sense to just drop it and use `setCommandScript` everywhere (otherwise this should also call `setCommandScriptContent` but unnecessary).